File: InMotionGIT.Common - Copy/Base/ActionCollection.vb
Imports System.Collections.ObjectModel
Imports System.Xml.Serialization

Namespace Base

    <Serializable()>
    <XmlType(Namespace:="urn:InMotionGIT.UI.Model")>
    <XmlRoot(Namespace:="urn:InMotionGIT.UI.Model")>
    Public Class ActionCollection
        Inherits Collection(Of Action)

        Public Function Clone() As ActionCollection
            Dim _actionCollection As New ActionCollection
            Dim newItem As Action

            For Each ruleItem As Action In Me
                newItem = ruleItem.Clone
                _actionCollection.Add(newItem)
            Next

            Return _actionCollection
        End Function

    End Class

End Namespace

File: Common/InMotionGIT.Common/Base/ActionCollection.cs
using System;
using System.Collections.ObjectModel;
using System.Xml.Serialization;

namespace InMotionGIT.Common.Base
{

    [Serializable()]
    [XmlType(Namespace = "urn:InMotionGIT.UI.Model")]
    [XmlRoot(Namespace = "urn:InMotionGIT.UI.Model")]
    public class ActionCollection : Collection<Action>
    {

        public ActionCollection Clone()
        {
            var _actionCollection = new ActionCollection();
            Action newItem;

            foreach (Action ruleItem in this)
            {
                newItem = ruleItem.Clone();
                _actionCollection.Add(newItem);
            }

            return _actionCollection;
        }

    }

}